hs_core/hydroshare/resource.py

```diff
@@ -957,17 +957,20 @@ def delete_format_metadata_after_delete_file(resource, file_name):
             resource.metadata.delete_element(format_element.term, format_element.id)
 
 
-def delete_resource_file(pk, filename_or_id, user):
+def delete_resource_file(pk, filename_or_id, user, delete_logical_file=True):
     """
-    Deletes an individual file from a HydroShare resource. If the file does not exist, the Exceptions.NotFound exception
-    is raised.
+    Deletes an individual file from a HydroShare resource. If the file does not exist,
+    the Exceptions.NotFound exception is raised.
 
     REST URL:  DELETE /resource/{pid}/files/{filename}
 
     Parameters:
     pk - The unique HydroShare identifier for the resource from which the file will be deleted
     filename - Name of the file to be deleted from the resource
     user - requesting user
+    delete_logical_file - If True then the ResourceFile object to be deleted if it is part of a
```

Comment:
@alvacouch 

> And the concept of logical file should not be limited to one kind of resource. It is needed everywhere.

I agree that logical file type should apply to all resource types. In later iterations of file types, this can be implemented (agile way).